### PR TITLE
feat: config-provider

### PR DESCRIPTION
**Pull request by GaoNeng-wWw**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated x(for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Add config-provider component

Issue Number: #226 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: examples/docs/resources/pc/demo/config-provider/text-direct.vue [hunk 1]
```diff
@@ -0,0 +1,135 @@
+<template>
+  <tiny-config-provider :design="design">
+    <h1>Text</h1>
+    <h1>المعرفة نور والجهل ظلام</h1>
```
Comment (kagol) [inline examples/docs/resources/pc/demo/config-provider/text-direct.vue]: @GaoNeng-wWw 这个是什么语言？为什么用这个语言呢？

File: packages/vue/src/config-provider/src/index.vue [hunk 2]
```diff
@@ -1,21 +1,69 @@
...
+    enable: true,
+    name: 'div'
+  },
+}
 export default defineComponent({
   name: $prefix + 'ConfigProvider',
   props: {
     design: {
-      type: Object
+      type: Object as PropType<ConfigProviderProps>,
```
Comment (zzcr) [inline packages/vue/src/config-provider/src/index.vue]: 这个design属性是我们适配多主题规范的作用，所以你可以重新定一个props去实现您的功能。

File: examples/docs/resources/pc/demo/config-provider/text-direct.vue (hunk 1 above)
Comment (GaoNeng-wWw) [reply examples/docs/resources/pc/demo/config-provider/text-direct.vue]: > @GaoNeng-wWw 这个是什么语言？为什么用这个语言呢？

这是阿拉伯语，阿拉伯语书写系统是RTL，所以就用了阿拉伯语

File: packages/vue/src/config-provider/src/index.vue (hunk 2 above)
Comment (GaoNeng-wWw) [reply packages/vue/src/config-provider/src/index.vue]: > 这个design属性是我们适配多主题规范的作用，所以你可以重新定一个props去实现您的功能。

收到

**Review by zzcr — APPROVED** [review]

**Review by zzcr — APPROVED** [review]

**Pull request merged**